---
title: "Untitled"
output: 
  flexdashboard::flex_dashboard:
    orientation: columns
    vertical_layout: fill
---

```{r setup, include=FALSE}
library(flexdashboard)
```

Column {data-width=650}
-----------------------------------------------------------------------

### Chart A

```{r}
library(ggplot2)
library(shiny)
pacman::p_load(
  rio,          # file import/export
  here,         # relative filepaths 
  lubridate,    # working with dates/epiweeks
  aweek,        # alternative package for working with dates/epiweeks
  incidence2,   # epicurves of linelist data
  i2extras,     # supplement to incidence2
  stringr,      # search and manipulate character strings
  forcats,      # working with factors
  RColorBrewer, # Color palettes from colorbrewer2.org
  tidyverse     # data management + ggplot2 graphics
)
linelist<-import(here("linelist_cleaned.xlsx"))
central_data <- linelist %>% 
  filter(hospital == "Central Hospital")

my_labels<-as_labeller(c(
"0-4"="Ages 0-4",
     "5-9"="Ages 5-9",
     "10-14"="Ages 10-14",
     "15-19"="Ages 15-19",
     "20-29"="Ages 20-29",
     "30-49"="Ages 30-49",
     "50-69"="Ages 50-69",
     "70+"="Over age 70"))
weekly_breaks <- seq.Date(from = as.Date("2014-02-01"),
                          to = as.Date("2015-07-15"),
                          by = "week")
# Sequence of weekly Monday dates for CENTRAL HOSPITAL
weekly_breaks_central <- seq.Date(from = as.Date(gsub(x=(floor_date(min(central_data$date_onset, na.rm=T),   "week", week_start = 1)),pattern=" UTC", replacement="", fixed=T)), # monday before first case
  to   = as.Date(gsub(x=(ceiling_date(max(central_data$date_onset, na.rm=T), "week", week_start = 1)),pattern=" UTC", replacement="",fixed=T)), # monday after last case
  by   = "week")

# make plot
###########
ggplot(central_data)+
	geom_histogram(
	mapping =aes(
	x =as.Date(date_onset),
      group =age_cat,
      fill =age_cat),    # arguments inside aes() apply by group
	
	color ="black",   # arguments outside aes() apply to all data
	# histogram breaks
	breaks =weekly_breaks_central, # pre-defined date vector (see earlier in this page)
	closed ="left"# count cases from start of breakpoint
	)+
	
	# The labels on the x-axis
	scale_x_date(
	expand            =c(0,0),  # remove excess x-axis space below and after case bars
	date_breaks       ="2 months",     # labels appear every 2 months
	date_minor_breaks ="1 month",      # vertical lines appear every 1 month 
	date_labels       ="%b\n'%y")+# date labels format
	# y-axis
	scale_y_continuous(expand =c(0,0))+# removes excess y-axis space between bottom of bars and the labels
	# aesthetic themes
	theme_minimal()+ # a set of themes to simplify plot
	theme(
	plot.caption =element_text(face ="italic", hjust =0), # caption on left side in italics
	axis.title =element_text(face ="bold"),
    legend.position ="bottom",
    strip.text =element_text(face ="bold", size =10),
    strip.background =element_rect(fill ="grey"))+# axis titles in bold
	# create facets
	facet_wrap(
	~age_cat,
    ncol =4,
    strip.position ="top",
    labeller =my_labels)+
	# labels
	labs(
	title    ="Weekly incidence of cases, by age category",
    subtitle ="Subtitle",
    fill     ="Age category",                                      # provide new title for legend
	x        ="Week of symptom onset",
    y        ="Weekly incident cases reported",
    caption  =stringr::str_glue("n = {nrow(central_data)} from Central Hospital; Case onsets range from {format(as.Date(min(central_data$date_onset, na.rm=T)), format = '%a %d %b %Y')} to {format(as.Date(max(central_data$date_onset, na.rm=T)), format = '%a %d %b %Y')}\n{nrow(central_data %>% filter(is.na(date_onset)))} cases missing date of onset and not shown"))

```

### Data

```{r}
variants_data<-import(here("phewas-catalog_karla.xlsx"))

names(variants_data)[names(variants_data) == "chromosome"] <- "chr"
names(variants_data)[names(variants_data) == "snp"] <- "snp"
names(variants_data)[names(variants_data) == "cases"] <- "Cases"
names(variants_data)[names(variants_data) == "phewas phenotype"] <- "Phenotype"
names(variants_data)[names(variants_data) == "p-value"] <- "P-vlue"
names(variants_data)[names(variants_data) == "gene_name"] <- "Gene name"
names(variants_data)[names(variants_data) == "gwas-associations"] <- "Gwas association"
DT::datatable(variants_data[, c("chr","snp","Gene name","Phenotype","Cases","P-value", "Gwas association")],
  rownames = FALSE, options = list(pageLength = 10)
)
```

Column {data-width=350}
-----------------------------------------------------------------------

### Chart B

```{r}
library(rnaturalearth)
library(wbstats)
library(leaflet)
library(DT)
map <- ne_countries()
names(map)[names(map) == "iso_a3"] <- "ISO3"
names(map)[names(map) == "name"] <- "NAME"
d <- wb(indicator = "EN.ATM.PM25.MC.M3",startdate = 2016, enddate = 2016)
map$PM2.5 <- d[match(map$ISO3, d$iso3c), "value"]
DT::datatable(map@data[, c("ISO3", "NAME", "PM2.5")],
  rownames = FALSE, options = list(pageLength = 10)
)
```

### Chart C

```{r}
ggplot(data = map@data, aes(x = PM2.5)) + geom_histogram()
```

### Chart D
```{r}
library(leaflet)

pal <- colorBin(
  palette = "viridis", domain = map$PM2.5,
  bins = seq(0, max(map$PM2.5, na.rm = TRUE) + 10, by = 10)
)


map$labels <- paste0(
  "<strong> Country: </strong> ",
  map$NAME, "<br/> ",
  "<strong> PM2.5: </strong> ",
  map$PM2.5, "<br/> "
) %>%
  lapply(htmltools::HTML)

leaflet(map) %>%
  addTiles() %>%
  setView(lng = 0, lat = 30, zoom = 2) %>%
  addPolygons(
    fillColor = ~ pal(PM2.5),
    color = "white",
    fillOpacity = 0.7,
    label = ~labels,
    highlight = highlightOptions(
      color = "black",
      bringToFront = TRUE
    )
  ) %>%
  leaflet::addLegend(
    pal = pal, values = ~PM2.5,
    opacity = 0.7, title = "PM2.5"
  )
```
